gpkit/constraints/set.py

```diff
@@ -55,8 +55,19 @@ def __init__(self, constraints, substitutions=None):
                 if hasattr(self[i], "substitutions"):
                     self.substitutions.update(self[i].substitutions)
         self.reset_varkeys()
-        self.substitutions.update({self[k].key: v for k, v in subs.items()
-                                   if k in self.varkeys})
```

Comment:
let's keep the deleted lines above, commented out, with a note about timing? (I didn't realize the alternative implementation would be so many more lines. Totally open to moving back to @bqpd's original short and sweet version as long as it's not slow.)